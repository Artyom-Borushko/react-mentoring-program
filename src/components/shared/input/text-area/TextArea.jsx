import React from 'react';
import { useField } from 'formik';
import '../input.css';

export default function TextArea({ label, ...props }) {
  const [field, meta] = useField(props);

  return (
    <div className="input-wrapper textarea">
      <label htmlFor={props.id || props.name}>{label}</label>
      <textarea className="text-area" {...field} {...props} />
      {meta.touched && meta.error ? (
        <div className="form-field-validation-error">{meta.error}</div>
      ) : null}
    </div>
  );
}
