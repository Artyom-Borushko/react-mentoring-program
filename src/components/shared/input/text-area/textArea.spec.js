import { render, screen } from '@testing-library/react';
import React from 'react';
import '@testing-library/jest-dom';
import { Formik } from 'formik';
import userEvent from '@testing-library/user-event';
import TextArea from './TextArea';

describe('Text area input component tests', () => {
  it('renders text area input with Formik', () => {
    const { container } = render(
      <Formik
        initialValues={{ movieOverview: '' }}
        onSubmit={() => {}}
      >
        <TextArea
          label="Overview"
          name="movieOverview"
          id="movieOverview"
          placeholder="Movie description"
        />
      </Formik>,
    );

    expect(container).toMatchSnapshot();
  });

  it('can type in text area input with Formik', async () => {
    render(
      <Formik
        initialValues={{ movieOverview: '' }}
        onSubmit={() => {}}
      >
        <TextArea
          label="Overview"
          name="movieOverview"
          id="movieOverview"
          placeholder="Movie description"
        />
      </Formik>,
    );

    const descriptionInput = screen.getByLabelText('Overview');
    await userEvent.click(descriptionInput);
    await userEvent.keyboard('test movie description 123');

    expect(descriptionInput.value).toBe('test movie description 123');
  });
});
