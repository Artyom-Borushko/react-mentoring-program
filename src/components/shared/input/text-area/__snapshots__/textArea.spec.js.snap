// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text area input component tests renders text area input with Formik 1`] = `
<div>
  <div
    class="input-wrapper textarea"
  >
    <label
      for="movieOverview"
    >
      Overview
    </label>
    <textarea
      class="text-area"
      id="movieOverview"
      name="movieOverview"
      placeholder="Movie description"
    />
  </div>
</div>
`;
